watcher/lib/influxdb/influxdb.go

```diff
@@ -307,14 +316,25 @@ func backwardsCompatibleClient(config Config) (*client_v2.Client, error) {
 	return &client, nil
 }
 
-func createAdminUser(client client_v2.Client, config Config) error {
+func upsertAdminUser(client client_v2.Client, config Config) error {
 	log.Infof("Creating admin user %v", config.InfluxDBAdminUser)
 	query := fmt.Sprintf(createAdminQuery, config.InfluxDBAdminUser, config.InfluxDBAdminPassword)
 	response, err := client.Query(client_v2.NewQuery(query, "", ""))
 	if err != nil {
 		return trace.Wrap(err, "failed to create admin user")
 	}
 	if response.Error() != nil {
+		if trace.IsAlreadyExists(ConvertError(response.Error())) {
```

Comment:
That's too much logic for an error case. First of all, I'd reverse the top-level condition and exit early on a happy path and move the rest of the handling for the case when the user already exists to a new function::
```go
if response.Error() == nil {
	return nil
}
if trace.IsAlreadyExists(ConvertError(response.Error())) {
	return updateUserPassword(...)
}
return trace.Wrap(response.Error(), "failed to create admin user")
```